docs2/site/docs/getting-started/transport.md

```diff
@@ -0,0 +1,85 @@
+
+**Note:** the *Query* property can be null in case of [persisted queries](https://www.apollographql.com/docs/apollo-server/performance/apq/) when a client sends only SHA-256 hash of the query in *Extensions* given that corresponding key-value pair has been saved on a server beforehand
+
+## OperationMessage
+
+`OperationMessage` represents a message typically used by the [GraphQL-WS](https://the-guild.dev/graphql/ws) or  [graphql-transport-ws](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) WebSockets-based protocols.
+Both of these protocols are used by the [Apollo Client](https://www.apollographql.com/docs/react/) library. The message contains the following properties:
+- Id - (Optional, string): The id of the message.
+- Type - (Required, string): The type of the message. Can be one of the following:
+    #### GraphQL-WS
+    - `connection_init` - Client -> Server. Client sends this message to initialize the connection.
+    - `connection_ack` - Server -> Client. Server sends this message to acknowledge the connection.
+    - `error` - Server -> Client. Server sends this message to indicate that an error occurred.
+    - `ping` - Bidirectional. Client or Server sends this message to keep the connection alive.
+    - `pong` - Bidirectional. Client or Server sends this message to keep the connection alive.
+    - `subscribe` - Client -> Server. Client sends this message to subscribe to a GraphQL subscription.
+    - `next` - Server -> Client. Server sends this message to indicate that a new value has been received for a GraphQL subscription.
+    - `complete` - Bidirectional. This message indicate that the GraphQL subscription has completed.
+
+    #### graphql-transport-ws
+    - `connection_init` - Client -> Server. Client sends this message to initialize the connection.
+    - `connection_ack` - Server -> Client. Server sends this message to acknowledge the connection.
+    - `connection_error` - Server -> Client. Server sends this message to indicate that an error occurred while establishing the connection.
+    - `ka` - Server -> Client. Server sends this message periodically to keep the client connection alive.
+    - `connection_terminate` - Client -> Server. Client sends this message to terminate the connection.
+    - `start` - Client -> Server. Client sends this message to start a GraphQL operation.
+    - `data` - Server -> Client. Server sends this message to send the GraphQL execution result.
+    - `error` - Server -> Client. Server sends this message to indicate that an error occurred during the GraphQL operation.
+    - `complete` - Server -> Client. Server sends this message to indicate that the GraphQL operation is complete.
+    - `stop` - Client -> Server. Client sends this message to stop a running GraphQL operation.
+- Payload - (Optional, object): The payload of the message. The payload is only used for *subscribe*, *GQL_DATA*, *GQL_ERROR*, and *GQL_COMPLETE* messages.
```

Comment:
This should probably be updated since we've replaced the constants with the actual values.  It might also be good to mention the data type required for start/subscribe messages (GraphQLRequest) and data/error messages (ExecutionResult).

Reply:
Everything else looks great!

Reply:
```suggestion
- Payload - (Optional, object): The payload of the message. The payload is only used for *subscribe*, *data*, *error*, and *complete* type messages. For client side messages, the payload is a `GraphQLRequest`. For server messages, the payload is an `ExecutionResult`. 
```

Are all messages that use payload mentioned in here?

Reply:
Well the `connection_init` message may pass arbitrary data, usually used for authentication information when connecting from a browser.  `connection_ack` may pass arbitrary data also.  For GraphQL-WS, `error` has an `ExecutionError` rather than an `ExecutionResult` instance.  But for `subscriptions-transport-ws`, `error` has an entire `ExecutionResult`.  `ping` and `pong` may also transfer arbitrary data.

Reply:
It may not be so important to list all the payload details here, either.  GraphQL.NET Server will take care of these details for the most part.  It might suffice to say "the payload is dependent on the message, typically a `GraphQLRequest` or `ExecutionResult` instance", or something along those lines.